components/security/core/authentication.rst

```diff
@@ -0,0 +1,229 @@
+worries, you'll fill the missing variables within a couple of minutes)::
+
+    use Symfony\Component\Security\Core\Exception\AuthenticationException;
+
+    try {
+        $authenticatedToken = $authManager->authenticate($unauthenticatedToken);
+    } catch (AuthenticationException $e) {
+        // ... authentication failed, do something
+    }
+
+The Security component comes with one very flexible authentication manager, the
+:class:`Symfony\\Component\\Security\\Core\\Authentication\\AuthenticationProviderManager`.
+This uses authentication providers to authenticate a token.
+
+.. _component-security-core-authentication-providers:
+
+Authentication Providers
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+One ``AuthenticationProviderManager`` can have multiple
+:class:`authentication providers <Symfony\\Component\\Security\\Core\\Authentication\\Provider\\AuthenticationProviderInterface>`.
+Using the ``supports()`` method, providers determines if they can authenticate
+the passed token.
+
+The most simple authentication provider is the
+``AnonymousAuthenticationProvider``. This providers "authenticates" an
+``AnonymousToken``. Anonymous users are guests, they aren't registered for the
+application, but they are still authenticated.
+
+.. code-block:: php
+
+    // ...
+    use Symfony\Component\Security\Core\Authentication\AuthenticationProviderManager;
+    use Symfony\Component\Security\Core\Authentication\Provider\AnonymousAuthenticationProvider;
+    use Symfony\Component\Security\Core\Authentication\Token\AnonymousToken;
+
+    $anonToken = new AnonymousToken('a_custom_key', 'Guest'.uniqid());
+
+    // 'a_custom_key' has to match the 1st argument of AnonymousToken.
+    $anonymousAuthProvider = new AnonymousAuthenticationProvider('a_custom_key');
+
+    $authManager = new AuthenticationProviderManager(array(
+        $anonymousAuthProvider,
+    ));
+
+    try {
+        $authenticatedToken = $authManager->authenticate($unauthenticatedToken);
+    } catch (AuthenticationException $e) {
+        // ... authentication failed, do something
+    }
+
+    // user is authenticated as a guest
+
+In this example, each requests to the application will result in an
+authenticated anonymous token with different names starting with ``Guest``.
+
+The DaoAuthenticationProvider (Data Access Object)
+..................................................
+
+This authentication provider is one of the most common authentication
+providers. It'll authenticate an ``UsernamePasswordToken``, retrieve the user,
+check the password and then return an authenticated ``UsernamePasswordToken``
+containing the user object. That's quite a lot, the following paragraphs each
+explain one of these steps.
+
+The first step is to retrieve the user. The dao provider does this with the
+help of a
+:class:`user provider <Symfony\\Component\\Security\\Core\\User\\UserProviderInterface>`.
+This can provide users from a database for example. The most basic user
```

Comment:
It can **provide** users from a database ... --> It can **fetch** users from a database ...